### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by arifjehoh**
# See in the `Files changed` tab

What the files implemented for own testing.

1. Run `docker-compose up --build`
2. Run on separate terminal `docker exec -i ais_db_1 psql -U ais_dev < scripts/vagrant/ais-developer-database.sql` make sure you have the correct db in vagrant directory. This command will update the current db with the new one.
3. You might want to rewrite the `mange.py` row 6 where it says `production` to `local`
4. Close the existing docker container.
5. Run `docker-compose up`

### CODE REVIEW

**Comment by arifjehoh** [comment]
Does it work for:
- [ ] Windows
- [ ] Mac
- [ ] Ubuntu

**Comment by arifjehoh** [comment]
For Windows user I was able to get docker to work with these steps:

1. Install and activate `WSL2` (follow this [link](https://docs.microsoft.com/en-us/windows/wsl/install-win10#manual-installation-steps))
2. Install `Docker `(follow this [link](https://docs.docker.com/docker-for-windows/install/))
3. Download the repository and checkout to branch `arifjehoh/feature/docker` 
4. Open `Powershell` and run `docker-compose up --build`
5. Wait for the installation and if you get an error like 
```
...
Is the server running on host "db" (172.18.0.2) and accepting
ais_1  |        TCP/IP connections on port 5432?
...
```
6. Open an other terminal with `Bash` and change directory to the repository and execute this command. (Make sure you have downloaded the latest database file from Google Drive and save it under scripts/vagrant/)
```bash

docker exec -i ais_db_1 psql -U ais_dev < scripts/vagrant/ais-developer-database.sql
```
7. Turn off the docker by entering the keys `Control + c` 
8. Restart the docker by entering this command in `Powershell`: `docker-compose up --build`

**Review by mattiasstahre — DISMISSED** [review]
Worked for me.

**Comment by CreatlV** [comment]
You need to put the database initialization file:  https://drive.google.com/file/d/1qKO96mlUE1fTbU7DH1csDZfdNdYZ3tt0/view?usp=sharing as "init.sql" in the src folder. Also, after creating the containers for the first time, the docker has to be restarted as the db connection will fail.

**Review by arifjehoh — COMMENTED** [review]
[Some reading for startup order, where db start first and then the application](https://docs.docker.com/compose/startup-order/)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM python:3
+
+ENV PYTHONUNBUFFERED 1
+
+RUN apt-get update -y
+RUN apt-get -y install binutils libproj-dev gdal-bin
```
Comment (arifjehoh) [inline Dockerfile:6]: Add `nodejs` and `npm`

**Comment by mrporsev** [comment]
I got it working on Windows 10 by following these steps created by @arifjehoh:

1. Install WSL2 and Docker Desktop
2. Pull this branch: **arifjehoh/feature/docker** and rewrite the mange.py row 6 where it says production to local.
3. Open up the Docker Desktop application (very important before doing step 4)
4. Run powershell as administrator and change directory to your local ais repository and run the following command: ``` docker-compose -f docker-compose.yaml build ``` , If it runs without failure continue to step 5 otherwise ctrl c and retry the command in powershell.
5.  Open command prompt (cmd) as administrator and cd to local ais repository. Run the following command: ``` docker exec -i ais_db_1 psql -U ais_dev < scripts\vagrant\ais-developer-database.sql ```. If you get an error on this step it might be because you don't have the ais-developer-database.sql file on your local ais repository. If you do, make sure it is placed under scripts\vagrant\.

**Pull request closed without merging**